### PR TITLE
MGMT-9568: use client.update_host instead of update_hosts

### PR DESCRIPTION
**Pull request by danielerez**
* Moved update_hosts from utils to day2_cluster.
* Used client.update_host instead of deprecated client.update_hosts.

### CODE REVIEW

File: src/assisted_test_infra/test_infra/helper_classes/day2_cluster.py [hunk 1]
```diff
@@ -243,9 +243,34 @@ def set_nodes_hostnames_if_needed(self, network_name: str):
...
+        """
+        Update names of the hosts in a cluster from a dictionary of libvirt nodes.
+
+        An entry from the dictionary is matched to a host by the host's MAC address (of any NIC).
+        Entries that do not match any host in the cluster are ignored.
+
+        Arguments:
+            client -- An assisted service client
+            cluster_id -- ID of the cluster to update
+            libvirt_nodes -- A dictionary that may contain data about cluster hosts
```
Comment (osherdp) [inline src/assisted_test_infra/test_infra/helper_classes/day2_cluster.py:258]: We usually do:
```suggestion
        Args:
            client: An assisted service client
            cluster_id: ID of the cluster to update
            libvirt_nodes: A dictionary that may contain data about cluster hosts
```

File: src/assisted_test_infra/test_infra/helper_classes/day2_cluster.py [hunk 2]
```diff
@@ -243,9 +243,34 @@ def set_nodes_hostnames_if_needed(self, network_name: str):
...
+        """
+
```
Comment (osherdp) [inline src/assisted_test_infra/test_infra/helper_classes/day2_cluster.py:260]:
```suggestion
```

File: src/assisted_test_infra/test_infra/helper_classes/day2_cluster.py [hunk 3]
```diff
@@ -243,9 +243,34 @@ def set_nodes_hostnames_if_needed(self, network_name: str):
                 "to work around libvirt for Terrafrom not setting hostnames of IPv6 hosts",
                 self.config.cluster_id,
             )
-            utils.update_hosts(
-                self.api_client, self.config.cluster_id, libvirt_nodes, update_roles=False, update_hostnames=True
-            )
+            self.update_hosts(self.api_client, self.config.cluster_id, libvirt_nodes)
+
+    def update_hosts(self, client, cluster_id, libvirt_nodes):
```
Comment (osherdp) [inline src/assisted_test_infra/test_infra/helper_classes/day2_cluster.py:248]: Please add type annotations

**Review by osherdp — CHANGES_REQUESTED** [review]

File: src/assisted_test_infra/test_infra/helper_classes/day2_cluster.py (hunk 1 above)
Comment (danielerez) [reply src/assisted_test_infra/test_infra/helper_classes/day2_cluster.py:258]: right, just copy-pasted:) fixed

File: src/assisted_test_infra/test_infra/helper_classes/day2_cluster.py (hunk 2 above)
Comment (danielerez) [reply src/assisted_test_infra/test_infra/helper_classes/day2_cluster.py:260]: done

File: src/assisted_test_infra/test_infra/helper_classes/day2_cluster.py (hunk 3 above)
Comment (danielerez) [reply src/assisted_test_infra/test_infra/helper_classes/day2_cluster.py:248]: done

**Review by osherdp — APPROVED** [review]

**Comment by danielerez** [comment]
rebased

**Comment by danielerez** [comment]
> 

rebased

**Review by osherdp — APPROVED** [review]

**Pull request merged**